src/os_win/os_thread.c

```diff
@@ -33,10 +33,15 @@ __wt_thread_join(WT_SESSION_IMPL *session, wt_thread_t tid)
 {
 	WT_DECL_RET;
 
-	if ((ret = WaitForSingleObject(tid, INFINITE)) == WAIT_OBJECT_0)
-		return (0);
+	if ((ret = WaitForSingleObject(tid, INFINITE)) != WAIT_OBJECT_0)
+		/*
+		 * If we fail to wait, we will leak handles so do not continue
+		 */
+		WT_PANIC_RET(session, WT_PANIC, "Wait for thread join failed");
+
+	CloseHandle(tid);
 
-	WT_RET_MSG(session, ret, "WaitForSingleObject");
+	return (0);
```

Comment:
@markbenvenuto:

- Should we do something with "ret", maybe use it as the second argument to the `WT_PANIC_RET` call so we get an idea why `WaitForSingleObject` failed? If not, then I think we can remove the use of `ret` in this function.

- Also, it looks to me like `CloseHandle` can fail ([see here](https://msdn.microsoft.com/en-us/library/windows/desktop/ms724211(v=vs.85).aspx)), should we check and panic in that case as well?